### PR TITLE
AOB-311: Add a new attribute header cell type

### PR DESCRIPTION
**Pull request by phaseinducer**
This PR allows to customize header cells in the datagrid.
It gets the header attribute label from the database instead of translation file.

**Usage :**
```
datagrid:
    my-grid:
        ...
        columns:
            some_attribute_column:
                ...
                headerCell: attribute
                attributeCode: some_attribute_code
```

**Note :**

We need to validate the use of `catalogLocale` vs `uiLocale`

### CODE REVIEW

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                    throw new Error('The option "attributeCode" must be set for a header cell of type "attribute".');
+                }
+
+                this.attributeCode = options.column.get('extraOptions').attributeCode;
+
+                HeaderCell.prototype.initialize.apply(this, arguments);
+            },
+
+            render: function () {
+                FetcherRegistry.getFetcher('attribute').fetch(this.attributeCode)
```
Comment (pierallard) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js]: If you can, please don't fetch the data into the render() method and move it in the configure() method

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+define(
```
Comment (pierallard) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js]: Can you transform this file into .typescript and put a description on it ?

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js (hunk 1 above)
Comment (ValentinMerlet) [reply src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js]: You're right, we'll fix it.

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js (hunk 2 above)
Comment (ValentinMerlet) [reply src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js]: Sure

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js (hunk 1 above)
Comment (ValentinMerlet) [reply src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js]: `configure()` is not called however `HeaderCell` extends Backbone.View so it should be called. Any idea ?

Comment (phaseinducer) [reply src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datagrid/header-cell/attribute-header-cell.js]: The cell views are not instantiated by our builder so `configure()` is never called.

**Pull request merged**